examples/dnsdoctor/dnsdoctor.lua

```diff
@@ -0,0 +1,77 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- DNS Doctoring : Rewrite DNS type A record to a private address for local clients
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local function nop() end
+
+local function get_domain(skb, off)
+	local name = ""
+	local i = 0
+
+	repeat
+		local d = skb:getbyte(off + i)
+		i = i + 1
+	until d == 0
+
+	return name .. skb:getstring(off, i-1), i
+end
+
+local function dnsdoctor_tg(skb, par, userdata)
+	local sep = userdata:find(" ")
+	local target_dns = userdata:sub(1, sep - 1)
+	local dst_ip, target_ip = string.unpack("I4I4", userdata:sub(sep + 1))
+	local thoff = par.thoff
+
+	local packetdst = skb:getuint32(16)
```

Comment:
you need to handle endianness here, right?

Reply:
I am sending the IP address from the userspace in the network order using `string.pack(">I4>I4", ... )`